### ISSUE TITLE
certain config results in a newline being inserted after completion

### ISSUE THREAD

**Issue by carlsverre** [ervandew/supertab#158]
I have a really weird bug that I can't nail down.  Basically when I use the following config and I press tab on a string which ends up completing to a single result after completing the word it inserts a newline.

```
let g:SuperTabDefaultCompletionType = "context"
let g:SuperTabContextDefaultCompletionType = "<c-n>"

let g:SuperTabCompletionContexts = ["s:ContextText"]
let g:SuperTabContextTextFileTypeExclusions = ["javascript", "javascript.jsx"]

let g:SuperTabLongestEnhanced = 1
let g:SuperTabLongestHighlight = 1
```

If I set `g:SuperTabLongestHighlight` to 0 then the bug is fixed.  It seems be something to do with the additional highlight logic interacting with the single result in a weird way.

Other than that, supertab is awesome!  Nice job :)

**Comment by kminh** [comment]
The culprit seems to be this commit: 70f4e9ce60dfbe9b323f33e4585cfaedb850f3e8

I try reverting to the commit right before that one and the issue is gone.

**Comment by Corax26** [comment]
I also experience this (quite annoying) issue, @kminh's workaround works for me too (thanks!).

**Comment by carlsverre** [comment]
@ervandew are you able to repro this issue as well?  Maybe the "autocmd CompleteDone" is being fired too early?

**Comment by timss** [comment]
Seems to be fixed in cdaa5c2 (https://github.com/ervandew/supertab/issues/162#issuecomment-263160209).

**Comment by Corax26** [comment]
Indeed, this is a duplicate of #162. I'm now using `let g:SuperTabLongestEnhanced = 1`again :)
